conans/model/build_info.py

```diff
@@ -225,6 +244,120 @@ def framework_paths(self):
         return self.frameworkdirs
 
 
+class DepCppInfo(object):
+
+    def __init__(self, cpp_info):
+        self._cpp_info = cpp_info
+        self._libs = None
+        self._system_libs = None
+        self._frameworks = None
+        self._defines = None
+        self._cxxflags = None
+        self._cflags = None
+        self._sharedlinkflags = None
+        self._exelinkflags = None
+
+        self._include_paths = None
+        self._lib_paths = None
+        self._bin_paths = None
+        self._build_paths = None
+        self._res_paths = None
+        self._src_paths = None
+        self._framework_paths = None
+        self._build_module_paths = None
+
+    def __getattr__(self, item):
+        try:
+            attr = self._cpp_info.__getattribute__(item)
+        except AttributeError:  # item is not defined, get config (CppInfo)
+            attr = self._cpp_info.__getattr__(item)
+        return attr
+
+    @staticmethod
+    def _merge_lists(seq1, seq2):
+        return [s for s in seq1 if s not in seq2] + seq2
+
+    def _aggregated_values(self, item):
+        if getattr(self, "_%s" % item) is None:
```

Comment:
Optimize this:
- If not components => no-op
- Read only once the ``getattr(self, "_%s" % item)``. If not None, return that value, do not read again
- If you need to set the value, return that value, do not do a ``getattr`` again